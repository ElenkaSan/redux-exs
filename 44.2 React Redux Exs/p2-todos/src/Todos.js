import React, { useState } from "react";
import { Button, Form, Input, Row, Col } from 'reactstrap';

function Todos({addTodo, updateTodos, deleteTodos, id}){
    const [update, setUpdate] = useState(addTodo);
    const [ifUpdate, setIfUpdate] = useState(false);

    function handleChange(e) {
        setUpdate(e.target.value);
    }
    function doUpdate() {
        setIfUpdate(ev => ! ev);
    }
    function handleDelete() {
        deleteTodos(id);
    }
    function Updated(e) {
        e.preventDefault();
        updateTodos(id, update);
        setIfUpdate(false);
    }

    return (
        (ifUpdate) ?
        <div>
        <Form onSubmit={Updated}>
        <Row>
        <Col className="p-2">
          <Input
            type="text"
            value={update}
            onChange={handleChange}
          />
          <Button className="btn btn-success" >Done</Button>
        </Col>
          </Row>
        </Form>
        </div>
        :
        <div className="col-md-6 row">
            <Col className="p-2">
            <li className="lead">{update}</li>
            <Button className="btn btn-warning m-2" onClick={doUpdate}>Edit</Button>
            <Button className="btn btn-danger" onClick={handleDelete}>Remove</Button>
            </Col>
        </div>
    )
}

export default Todos;