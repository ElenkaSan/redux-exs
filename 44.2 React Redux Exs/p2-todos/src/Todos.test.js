import React from 'react';
import Todos from './Todos';
import {render, fireEvent, screen} from '@testing-library/react';

it('check render without crashing', function() {
    render(<Todos/>);
});

it('passing snapshot test', function() {
    expect(render(<Todos addTodo ="wish list"/>).asFragment()).toMatchSnapshot();
});

it('does not show the form by default and toggles on click', function(){
    render(<Todos addTodo ="update todo" updateTodos={()=> undefined}/>);
    
    expect(screen.queryByDisplayValue("update todo")).not.toBeInTheDocument();
    const editButton = screen.getByText("Edit");
    expect(editButton).toBeInTheDocument();
    fireEvent.click(editButton);
    expect(screen.getByDisplayValue("update todo")).toBeInTheDocument();
    fireEvent.click(screen.getByText("Done"));

});