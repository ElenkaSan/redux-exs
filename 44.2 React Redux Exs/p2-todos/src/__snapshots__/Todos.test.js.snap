// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passing snapshot test 1`] = `
<DocumentFragment>
  <div
    class="col-md-6 row"
  >
    <div
      class="p-2 col"
    >
      <li
        class="lead"
      >
        wish list
      </li>
      <button
        class="btn btn-warning m-2 btn btn-secondary"
        type="button"
      >
        Edit
      </button>
      <button
        class="btn btn-danger btn btn-secondary"
        type="button"
      >
        Remove
      </button>
    </div>
  </div>
</DocumentFragment>
`;
